cnd/migrations/2020-05-06-105000_created-swap-tables/up.sql

```diff
@@ -0,0 +1,64 @@
+-- Your SQL goes here
+
+CREATE TABLE swaps
+(
+    id INTEGER			NOT NULL PRIMARY KEY,
+    local_swap_id UNIQUE	NOT NULL,
+    role			NOT NULL,
+    counterparty_peer_id	NOT NULL
+);
+
+CREATE TABLE secret_hashes
+(
+    id INTEGER			NOT NULL PRIMARY KEY,
+    swap_id INTEGER		NOT NULL,
+    secret_hash		NOT NULL,
+    FOREIGN KEY (swap_id)	REFERENCES swaps(id)
+);
+
+CREATE TABLE shared_swap_ids
+(
+    id INTEGER			NOT NULL PRIMARY KEY,
+    swap_id INTEGER		NOT NULL,
+    shared_swap_id		NOT NULL,
+    FOREIGN KEY (swap_id)	REFERENCES swaps(id)
+);
+
+CREATE TABLE address_hints
+(
+    id INTEGER			NOT NULL PRIMARY KEY,
+    swap_id INTEGER		NOT NULL,
+    address_hint		NOT NULL,
+    FOREIGN KEY(swap_id)	REFERENCES swaps(id)
```

Comment:
I would remove the reference to swap and store the Peer ID instead.

Reply:
I was unable to create a proper FK relationship betweek `swap.counterparty_peer_id` and `address_hint.counterparty_peer_id` - if we don't need this relationship that is fine. But that would mean, that we would always have to try to dial all `address_hints` a `peer_id` yields in this table (that could be named `address_book`)

Reply:
> But that would mean, that we would always have to try to dial all address_hints a peer_id yields in this table (that could be named address_book)

Yes that is why I would do.